samples/apps/video-streaming-app/streaming.py

```diff
@@ -19,13 +19,148 @@
+                sleep(1)
 
-def get_camera_list(configuration_name):
-    camera_list = []
+class CameraDisplay:
+    def __init__(self):
+        self.main_camera = None
+        self.small_cameras = []
+        self.mutex = threading.Lock()
+
+    def __eq__(self, other):
+        return self.main_camera == other.main_camera and self.small_cameras == other.small_cameras
+        
+    def StartHandlers(self):
+        if self.main_camera is not None:
+            self.main_camera.StartHandler()
+        for small_camera in self.small_cameras:
+            small_camera.StartHandler()
+
+    def WaitHandlers(self):
+        global global_stop_event
+
+        global_stop_event.set()
+        if self.main_camera is not None:
+            self.main_camera.WaitHandler()
+        for small_camera in self.small_cameras:
+            small_camera.WaitHandler()
+        global_stop_event.clear()
+        
+    def Merge(self, other):
+        self.mutex.acquire()
+        try:
+            self.WaitHandlers()
+
+            self.main_camera = other.main_camera
+            self.small_cameras = other.small_cameras
+
+            self.StartHandlers()
+        finally:
+            self.mutex.release()
+
+    def Count(self):
+        self.mutex.acquire()    
+        result = len(self.small_cameras)
+        if self.main_camera is not None:
+            result += 1
+        self.mutex.release()
+        return result
+
+    def HashCode(self):
+        self.mutex.acquire()
+        cameras = ",".join([camera.url for camera in self.small_cameras])
+        if self.main_camera is not None:
+            cameras = "{0}+{1}".format(self.main_camera.url, cameras)
+        self.mutex.release()
+        return cameras
+
+    def StreamFrames(self, camera_id):
+        selected_camera = None
+        camera_id = int(camera_id)
+
+        self.mutex.acquire()
+        if camera_id == 0:
+            selected_camera = self.main_camera
+        elif camera_id - 1 < len(self.small_cameras):
+            selected_camera = self.small_cameras[camera_id - 1]
+        self.mutex.release()
+        
+        if selected_camera is None:
+            return Response(None, 500)
+        else:
+            return Response(selected_camera.GeneratorFunc(), mimetype='multipart/x-mixed-replace; boundary=frame')
+
+def get_camera_display(configuration_name):
+    camera_display = CameraDisplay()
+    
     config.load_incluster_config()
     coreV1Api = client.CoreV1Api()
+
+    ret = coreV1Api.list_service_for_all_namespaces(watch=False)
+    for svc in ret.items:
+        if svc.metadata.name == configuration_name + "-svc":
+            grpc_ports = list(
+                filter(lambda port: port.name == "grpc", svc.spec.ports))
+            if (len(grpc_ports) == 1):
+                url = "{0}:{1}".format(svc.spec.cluster_ip, grpc_ports[0].port)
+                camera_display.main_camera = CameraFeed(url)
+
     ret = coreV1Api.list_service_for_all_namespaces(watch=False)

```

Comment:
can we consolidate the 2 for loops here?  or call list_service_for_all_namespaces once?

Reply:
Thanks, you are right. They were originally at two different places and I missed refactoring those. Coming up shortly.